app/src/main/cpp/LoadingAnimation.cpp

```diff
@@ -0,0 +1,93 @@
+/* -*- Mode: C++; tab-width: 20; indent-tabs-mode: nil; c-basic-offset: 2 -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "LoadingAnimation.h"
+#include "vrb/ConcreteClass.h"
+#include "vrb/CreationContext.h"
+#include "vrb/Matrix.h"
+#include "vrb/ModelLoaderAndroid.h"
+#include "vrb/TextureGL.h"
+#include "vrb/Toggle.h"
+#include "vrb/Transform.h"
+
+#include "Quad.h"
+
+namespace crow {
+
+struct LoadingAnimation::State {
+  vrb::CreationContextWeak context;
+  vrb::TextureGLPtr texture;
+  vrb::TogglePtr root;
+  QuadPtr spinner;
+  float rotation;
+
+  State()
+      : rotation(0.0f)
+  {}
+
+  void Initialize() {
+    vrb::CreationContextPtr create = context.lock();
+    root = vrb::Toggle::Create(create);
+  }
+};
+
+
+void
+LoadingAnimation::LoadModels(const vrb::ModelLoaderAndroidPtr& aLoader) {
+  vrb::LoadTask task = [this](vrb::CreationContextPtr& aContext) -> vrb::GroupPtr {
+    m.texture = aContext->LoadTexture("webvr_spinner.png", true);
+    return nullptr;
+  };
+
+  vrb::LoadFinishedCallback callback = [this](vrb::GroupPtr& aGroup){
+    if (m.texture) {
+      vrb::CreationContextPtr create = m.context.lock();
+      float aspect = (float)m.texture->GetWidth() / (float)m.texture->GetHeight();
+      float size = 0.25f;
+      m.spinner = Quad::Create(create, size * aspect, size);
+      m.spinner->SetTexture(m.texture, m.texture->GetWidth(), m.texture->GetHeight());
+      m.root->AddNode(m.spinner->GetRoot());
+    }
+  };
+
+  aLoader->RunLoadTask(nullptr, task, callback);
+}
+
+
+void
+LoadingAnimation::Update() {
```

Comment:
This reminds me, I want to update vrb::Updatable::UpdateResource() to get passed the delta time from last update so it can be used for animation nodes as well. https://github.com/MozillaReality/vrb/issues/34